From a857448fa29c2d52e609ee4f1b149ebe64bc14f1 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 07 Oct 2020 21:52:59 +0300
Subject: [PATCH] [SW-21702] goya2: remove pcie enable check

Since PCIe bringup will be only done as part of ppboot, this patch
removes pcie enable check as part of pcie link up flow.

Change-Id: I744746924d998c6f14d1df0bd6c951bc0b100588
---

diff --git a/goya2/common.c b/goya2/common.c
index 0ed61e7..43b7a05 100644
--- a/goya2/common.c
+++ b/goya2/common.c
@@ -133,22 +133,9 @@
 	return reset_cause;
 }
 
-/*
- * This logic copied from BTL PCIe section (#4)
- * In BTL it is called when DIP switch for PCIe enable is set
- * Now we wish to do this logic also here as a measure of security
- * It means that in new PCB's version by default PCIe won't be enabled
- * So without proper FW device won't be accessible through PCIe
- * Hence device is more secured
- */
 static void preboot_pcie_enable(void)
 {
-	u32 boot_strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L);
 	u32 reset_cause;
-
-	/* Check if already done by BTL */
-	if (boot_strap & 0x2)
-		return;
 
 	/* During hard reset triggered by LKD keep PCIe intact */
 	reset_cause = preboot_clr_flops_reset_cause();
